cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -389,12 +413,19 @@ public void add(DatasetDefinition def) {
     @Override
     public <T extends DatasetDefinition> T get(String datasetTypeName) {
       T def;
+      Id.DatasetType datasetTypeId = Id.DatasetType.from(namespaceId, datasetTypeName);
       if (registry.hasType(datasetTypeName)) {
         def = registry.get(datasetTypeName);
       } else {
-        DatasetTypeMeta typeMeta = datasets.getTypeMDS().getType(datasetTypeName);
+        DatasetTypeMeta typeMeta = datasets.getTypeMDS().getType(datasetTypeId);
         if (typeMeta == null) {
-          throw new IllegalArgumentException("Requested dataset type is not available: " + datasetTypeName);
+          // not found in the user namespace. Try finding in the system namespace
+          datasetTypeId = Id.DatasetType.from(Constants.SYSTEM_NAMESPACE, datasetTypeName);
```

Comment:
Probably not this PR, but maybe file a JIRA to consolidate how these are resolved, if possible.